ruby/spec/chapter04_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'spec_helper'
+require './chapter04'
+
+describe 'Chapter 4' do
+  let(:conn) { Redis.new(db: 15) }
+
+  let(:user0) {
+    {
+      id: 0,
+      name: "Jon",
+      fund: 100,
+      items: ["Longclaw", "Ghost"]
+    }
+  }
+
+  let(:user1) {
+    {
+      id: 1,
+      name: "Arya",
+      fund: 300,
+      items: ["Needle", "Frey Pies"]
+    }
+  }
+
+  before do
+    conn.flushdb
+  end
+
+  it "inializes DB" do
+    c = Chapter4::Client.new(conn, [user0, user1])
+
+    expect(c.user(0)).to eq({"name"=>"Jon", "fund"=>"100"})
+    expect(c.user(1)).to eq({"name"=>"Arya", "fund"=>"300"})
+
+    expect(c.inventory(0)).to eq(["Longclaw", "Ghost"])
```

Comment:
```
Failures:

  1) Chapter 4 inializes DB
     Failure/Error: expect(c.inventory(0)).to eq(["Longclaw", "Ghost"])

       expected: ["Longclaw", "Ghost"]
            got: ["Ghost", "Longclaw"]

       (compared using ==)
     # ./spec/chapter04_spec.rb:35:in `block (2 levels) in <top (required)>'
```

Reply:
You should use `match_array` instead of `eq`

```ruby
expect(c.inventory(0)).to match_array(["Longclaw", "Ghost"])
```

ref: https://stackoverflow.com/a/36934494

> Sets are implemented using a hash table and are deterministic. That doesn't mean, however, that the order you get from SMEMBERS is what you'd expect.
>
> If you need lexicographical ordering, use a Sorted Set in which all members have the same score (e.g. Zero).